src/fsharp/service/SemanticClassification.fs

```diff
@@ -116,61 +129,198 @@ module TcResolutionsExtensions =
-                    | Item.RecdField rfinfo, _, _, _, _, m when isRecdFieldMutable rfinfo ->
-                        add m SemanticClassificationType.MutableVar
-                    | Item.RecdField rfinfo, _, _, _, _, m when isFunction g rfinfo.FieldType ->
-                        add m SemanticClassificationType.Function
-                    | Item.RecdField EnumCaseFieldInfo, _, _, _, _, m ->
-                        add m SemanticClassificationType.Enumeration
-                    | Item.MethodGroup _, _, _, _, _, m ->
-                        add m SemanticClassificationType.Function
-                    // custom builders, custom operations get colored as keywords
+
+                    | (Item.Value vref), _, _, _, _, m ->
+                        if isValRefDisposable vref then
+                            add m SemanticClassificationType.DisposableValue
+                        elif Option.isSome vref.LiteralValue then
+                            add m SemanticClassificationType.Literal
+                        elif not vref.IsCompiledAsTopLevel then
+                            add m SemanticClassificationType.LocalValue
+                        else
+                            add m SemanticClassificationType.Value
+
+                    | Item.RecdField rfinfo, _, _, _, _, m ->
+                        match rfinfo with
+                        | EnumCaseFieldInfo ->
+                            add m SemanticClassificationType.Enumeration
+                        | _ ->
+                            if isRecdFieldMutable rfinfo then
+                                add m SemanticClassificationType.MutableRecordField
+                            elif isFunTy g rfinfo.FieldType then
+                                add m SemanticClassificationType.RecordFieldAsFunction
+                            else
+                                add m SemanticClassificationType.RecordField
+
+                    | Item.AnonRecdField(_, tys, idx, m), _, _, _, _, _ ->
+                        let ty = tys.[idx]
+
+                        // It's not currently possible for anon record fields to be mutable, but they can be ref cells
+                        if isRefCellTy g ty then
+                            add m SemanticClassificationType.MutableRecordField
+                        elif isFunTy g ty then
+                            add m SemanticClassificationType.RecordFieldAsFunction
+                        else
+                            add m SemanticClassificationType.RecordField
+
+                    | Item.Property (_, pinfo :: _), _, _, _, _, m ->
+                        if not pinfo.IsIndexer then
+                            add m SemanticClassificationType.Property
+                        
+                    | (Item.CtorGroup _ | Item.DelegateCtor _ | Item.FakeInterfaceCtor _), _, _, _, _, m ->
+                        add m SemanticClassificationType.Constructor
+
```

Comment:
This will need to break out:

* `Item.CtorGroup` will need to distinguish between a class and a struct
* The other two are TBD if they get treated specially or not